packages/kokkos-kernels/src/graph/KokkosGraph_Distance1Color.hpp

```diff
@@ -74,8 +74,16 @@ void graph_color_symbolic(
 
   gch->set_tictoc(handle->get_verbose());
 
-  color_view_type colors_out = color_view_type("Graph Colors", num_rows);
-
+  //printf("GraphColorHandle's colorView reference count: %d, and label: %s",gch->get_vertex_colors().use_count(),
+  //                                                                         gch->get_vertex_colors().label());
+  
+  color_view_type colors_out;// = color_view_type("Graph Colors", num_rows);
+  if(gch->get_vertex_colors().use_count() > 0){
+    colors_out = gch->get_vertex_colors();
+  } else {
+    colors_out = color_view_type("Graph Colors", num_rows);
```

Comment:
As far as I know, yes. This part of the file checks to see whether or not the view passed in (via gch->get_vertex_colors()) is initialized. If it is, we use that. If not, the view needs to be initialized. Previously, the view was always just initialized, which didn't allow for passing in a partial coloring.

Reply:
I mean, why not do this?
```cpp
using Kokkos::view_alloc;
using Kokkos::WithoutInitializing;
colors_out = color_view_type(view_alloc("Graph Colors", WithoutInitializing), num_rows);
```

Reply:
The initialization is not something I added to this file, I just added the if/else surrounding the initialization. I'm not familiar enough with the KokkosKernels coloring codebase to say confidently that this will not break anything, so I don't think I should make this change.